src/Core/src/Handlers/DatePicker/DatePickerHandler.iOS.cs

```diff
@@ -31,12 +29,19 @@ protected override void ConnectHandler(MauiDatePicker platformView)
 				}
 			}
 
+			platformView.EditingDidBegin += OnEditingDidBegin;
+			platformView.EditingDidEnd += OnEditingDidEnd;
+
 			base.ConnectHandler(platformView);
 		}
 
 		protected override void DisconnectHandler(MauiDatePicker platformView)
 		{
+			platformView.EditingDidBegin -= OnEditingDidBegin;
```

Comment:
This will cause the Picker to leak because DisconnectHandler is not always called, we can't trust this. 

@PureWeen  I guess we really need to figure how to properly have a way to get this to be called and we can remove events .

Reply:
I think you can remove these events. We already expose these through the Delegate `class DatePickerDelegate : MauiDatePickerDelegate`

Subscribing to these events, I'm pretty sure,  also removes the delegate which is going to break other things.

Reply:
You are right in the iOS DatePicker case. Applied changes.